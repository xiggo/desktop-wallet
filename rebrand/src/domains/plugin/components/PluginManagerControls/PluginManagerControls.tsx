import React from "react";

import { LayoutControls } from "@/app/components/LayoutControls";

interface PluginManagerControlsProperties {
	onSelectGridView: any;
	onSelectListView: any;
	selectedViewType: string;
}

export const PluginManagerControls = ({
	onSelectGridView,
	onSelectListView,
	selectedViewType = "grid",
}: PluginManagerControlsProperties) => (
	<div data-testid="PluginManagerControls" className="flex items-center">
		<LayoutControls
			onSelectGridView={onSelectGridView}
			onSelectListView={onSelectListView}
			selectedViewType={selectedViewType}
		/>
	</div>
);
