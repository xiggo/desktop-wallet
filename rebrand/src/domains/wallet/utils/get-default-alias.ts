import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
