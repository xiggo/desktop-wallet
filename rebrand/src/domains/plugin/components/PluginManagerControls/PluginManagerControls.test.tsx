import userEvent from "@testing-library/user-event";
import React from "react";

import { PluginManagerControls } from "./PluginManagerControls";
import { render, screen } from "@/utils/testing-library";

describe("PluginManagerControls", () => {
	it("should render", () => {
		const { asFragment } = render(<PluginManagerControls />);

		expect(screen.getByTestId("PluginManagerControls")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should toggle between list and grid", () => {
		const Component = () => {
			const [viewType, setViewType] = React.useState("grid");

			return (
				<div>
					<span data-testid="viewType">{viewType}</span>
					<PluginManagerControls
						selectedViewType={viewType}
						onSelectListView={() => setViewType("list")}
						onSelectGridView={() => setViewType("grid")}
					/>
				</div>
			);
		};

		const { asFragment } = render(<Component />);

		const gridIcon = screen.getByTestId("LayoutControls__grid--icon");
		const listIcon = screen.getByTestId("LayoutControls__list--icon");

		userEvent.click(listIcon);

		expect(screen.getByTestId("viewType")).toHaveTextContent("list");

		userEvent.click(gridIcon);

		expect(screen.getByTestId("viewType")).toHaveTextContent("grid");

		expect(asFragment()).toMatchSnapshot();
	});
});
