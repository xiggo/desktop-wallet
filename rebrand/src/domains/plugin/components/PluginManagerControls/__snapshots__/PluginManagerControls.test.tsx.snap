// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginManagerControls should render 1`] = `
<DocumentFragment>
  <div
    class="flex items-center"
    data-testid="PluginManagerControls"
  >
    <div
      class="flex items-center space-x-1"
    >
      <div
        data-testid="LayoutControls__list"
      >
        <div
          class="group"
        >
          <button
            class="ControlButton__ControlButtonStyled-sc-1l8cuc8-0 ckIXFQ ring-focus"
            data-testid="LayoutControls__list--icon"
          >
            <div
              class="sc-bdvvtL qGXhc"
              height="20"
              width="20"
            >
              <svg>
                list.svg
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        data-testid="LayoutControls__grid"
      >
        <div
          class="group"
        >
          <button
            class="ControlButton__ControlButtonStyled-sc-1l8cuc8-0 ckIXFQ ring-focus active"
            data-testid="LayoutControls__grid--icon"
          >
            <div
              class="sc-bdvvtL qGXhc"
              height="20"
              width="20"
            >
              <svg>
                grid.svg
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PluginManagerControls should toggle between list and grid 1`] = `
<DocumentFragment>
  <div>
    <span
      data-testid="viewType"
    >
      grid
    </span>
    <div
      class="flex items-center"
      data-testid="PluginManagerControls"
    >
      <div
        class="flex items-center space-x-1"
      >
        <div
          data-testid="LayoutControls__list"
        >
          <div
            class="group"
          >
            <button
              class="ControlButton__ControlButtonStyled-sc-1l8cuc8-0 ckIXFQ ring-focus"
              data-testid="LayoutControls__list--icon"
            >
              <div
                class="sc-bdvvtL qGXhc"
                height="20"
                width="20"
              >
                <svg>
                  list.svg
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          data-testid="LayoutControls__grid"
        >
          <div
            class="group"
          >
            <button
              class="ControlButton__ControlButtonStyled-sc-1l8cuc8-0 ckIXFQ ring-focus active"
              data-testid="LayoutControls__grid--icon"
            >
              <div
                class="sc-bdvvtL qGXhc"
                height="20"
                width="20"
              >
                <svg>
                  grid.svg
                </svg>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
